packages/Webkul/API/Http/Controllers/Shop/SessionController.php

```diff
@@ -103,23 +106,19 @@ public function update()
             'password'      => 'confirmed|min:6',
         ]);
 
-        $data = request()->only('first_name','last_name','gender','date_of_birth','email','password');
-
-        if (! $data['date_of_birth']) {
-            unset($data['date_of_birth']);
-        }
+        $data = request()->only('first_name', 'last_name', 'gender', 'date_of_birth', 'email', 'password');
 
         if (!isset($data['password']) || ! $data['password']) {
             unset($data['password']);
         } else {
             $data['password'] = bcrypt($data['password']);
         }
 
-        $this->customerRepository->update($data, $customer->id);
+        $this->updatedCustomer->update($data, $customer->id);
```

Comment:
Only here,
`$updatedCustomer = $this->customerRepository->update($data, $customer->id);`